File: csharp/src/Stakkd/Stakkd.model/SkipTracingLookupResponseNamesAliases.cs
/*
 * Stakkd API
 *
 * # Introduction The Stakkd platform provides comprehensive data services that meet the business objectives of organizations ranging from Fortune 500 companies to startups. Our clients work in an array of industries, including insurance, financial, legal, travel, hospitality, retail, health, media, and telecommunications.  The Stakkd API is documented using the Open API 3.0.1 standard.  All endpoints are available at https://api.stakkd.io/v1. Detailed documentation for each endpoint is below, including the full URL for the endpoint.  # Account Types The API supports two different account types, each with different permissions.  ## Paid Account Paid accounts provide unlimited access to all of the URIs.  ## Trial Account Trial Accounts have a limited number of queries to the APIs. Keep in mind that queries which do not return results still count against the Trial Account query limits. Trial Accounts are temporary, but can be upgraded or extended by contacting [support@stakkd.io](mailto:support@stakkd.io) 
 *
 * The version of the OpenAPI document: v1
 * Generated by: https://github.com/openapitools/openapi-generator.git
 */


using System;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.IO;
using System.Runtime.Serialization;
using System.Text;
using System.Text.RegularExpressions;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using Newtonsoft.Json.Linq;
using System.ComponentModel.DataAnnotations;
using OpenAPIDateConverter = Stakkd.Client.OpenAPIDateConverter;

namespace Stakkd.Stakkd.model
{
    /// <summary>
    /// SkipTracingLookupResponseNamesAliases
    /// </summary>
    [DataContract(Name = "skipTracingLookupResponse_names_aliases")]
    public partial class SkipTracingLookupResponseNamesAliases : IEquatable<SkipTracingLookupResponseNamesAliases>, IValidatableObject
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="SkipTracingLookupResponseNamesAliases" /> class.
        /// </summary>
        /// <param name="lastName">The last name for the person..</param>
        /// <param name="middleName">The middle name for the person..</param>
        /// <param name="firstName">The first name for the person..</param>
        public SkipTracingLookupResponseNamesAliases(string lastName = default(string), string middleName = default(string), string firstName = default(string))
        {
            this.LastName = lastName;
            this.MiddleName = middleName;
            this.FirstName = firstName;
        }

        /// <summary>
        /// The last name for the person.
        /// </summary>
        /// <value>The last name for the person.</value>
        [DataMember(Name = "lastName", EmitDefaultValue = false)]
        public string LastName { get; set; }

        /// <summary>
        /// The middle name for the person.
        /// </summary>
        /// <value>The middle name for the person.</value>
        [DataMember(Name = "middleName", EmitDefaultValue = false)]
        public string MiddleName { get; set; }

        /// <summary>
        /// The first name for the person.
        /// </summary>
        /// <value>The first name for the person.</value>
        [DataMember(Name = "firstName", EmitDefaultValue = false)]
        public string FirstName { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.Append("class SkipTracingLookupResponseNamesAliases {\n");
            sb.Append("  LastName: ").Append(LastName).Append("\n");
            sb.Append("  MiddleName: ").Append(MiddleName).Append("\n");
            sb.Append("  FirstName: ").Append(FirstName).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }

        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return Newtonsoft.Json.JsonConvert.SerializeObject(this, Newtonsoft.Json.Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as SkipTracingLookupResponseNamesAliases);
        }

        /// <summary>
        /// Returns true if SkipTracingLookupResponseNamesAliases instances are equal
        /// </summary>
        /// <param name="input">Instance of SkipTracingLookupResponseNamesAliases to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(SkipTracingLookupResponseNamesAliases input)
        {
            if (input == null)
            {
                return false;
            }
            return 
                (
                    this.LastName == input.LastName ||
                    (this.LastName != null &&
                    this.LastName.Equals(input.LastName))
                ) && 
                (
                    this.MiddleName == input.MiddleName ||
                    (this.MiddleName != null &&
                    this.MiddleName.Equals(input.MiddleName))
                ) && 
                (
                    this.FirstName == input.FirstName ||
                    (this.FirstName != null &&
                    this.FirstName.Equals(input.FirstName))
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                if (this.LastName != null)
                {
                    hashCode = (hashCode * 59) + this.LastName.GetHashCode();
                }
                if (this.MiddleName != null)
                {
                    hashCode = (hashCode * 59) + this.MiddleName.GetHashCode();
                }
                if (this.FirstName != null)
                {
                    hashCode = (hashCode * 59) + this.FirstName.GetHashCode();
                }
                return hashCode;
            }
        }

        /// <summary>
        /// To validate all properties of the instance
        /// </summary>
        /// <param name="validationContext">Validation context</param>
        /// <returns>Validation Result</returns>
        public IEnumerable<System.ComponentModel.DataAnnotations.ValidationResult> Validate(ValidationContext validationContext)
        {
            yield break;
        }
    }

}

File: android/src/main/java/com/stakkd/model/SkipTracingLookupResponseNamesAliases.java
/**
 * Stakkd API
 * # Introduction The Stakkd platform provides comprehensive data services that meet the business objectives of organizations ranging from Fortune 500 companies to startups. Our clients work in an array of industries, including insurance, financial, legal, travel, hospitality, retail, health, media, and telecommunications.  The Stakkd API is documented using the Open API 3.0.1 standard.  All endpoints are available at https://api.stakkd.io/v1. Detailed documentation for each endpoint is below, including the full URL for the endpoint.  # Account Types The API supports two different account types, each with different permissions.  ## Paid Account Paid accounts provide unlimited access to all of the URIs.  ## Trial Account Trial Accounts have a limited number of queries to the APIs. Keep in mind that queries which do not return results still count against the Trial Account query limits. Trial Accounts are temporary, but can be upgraded or extended by contacting [support@stakkd.io](mailto:support@stakkd.io) 
 *
 * The version of the OpenAPI document: v1
 * 
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

package com.stakkd.model;

import io.swagger.annotations.*;
import com.google.gson.annotations.SerializedName;

@ApiModel(description = "")
public class SkipTracingLookupResponseNamesAliases {
  
  @SerializedName("lastName")
  private String lastName = null;
  @SerializedName("middleName")
  private String middleName = null;
  @SerializedName("firstName")
  private String firstName = null;

  /**
   * The last name for the person.
   **/
  @ApiModelProperty(value = "The last name for the person.")
  public String getLastName() {
    return lastName;
  }
  public void setLastName(String lastName) {
    this.lastName = lastName;
  }

  /**
   * The middle name for the person.
   **/
  @ApiModelProperty(value = "The middle name for the person.")
  public String getMiddleName() {
    return middleName;
  }
  public void setMiddleName(String middleName) {
    this.middleName = middleName;
  }

  /**
   * The first name for the person.
   **/
  @ApiModelProperty(value = "The first name for the person.")
  public String getFirstName() {
    return firstName;
  }
  public void setFirstName(String firstName) {
    this.firstName = firstName;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    SkipTracingLookupResponseNamesAliases skipTracingLookupResponseNamesAliases = (SkipTracingLookupResponseNamesAliases) o;
    return (this.lastName == null ? skipTracingLookupResponseNamesAliases.lastName == null : this.lastName.equals(skipTracingLookupResponseNamesAliases.lastName)) &&
        (this.middleName == null ? skipTracingLookupResponseNamesAliases.middleName == null : this.middleName.equals(skipTracingLookupResponseNamesAliases.middleName)) &&
        (this.firstName == null ? skipTracingLookupResponseNamesAliases.firstName == null : this.firstName.equals(skipTracingLookupResponseNamesAliases.firstName));
  }

  @Override
  public int hashCode() {
    int result = 17;
    result = 31 * result + (this.lastName == null ? 0: this.lastName.hashCode());
    result = 31 * result + (this.middleName == null ? 0: this.middleName.hashCode());
    result = 31 * result + (this.firstName == null ? 0: this.firstName.hashCode());
    return result;
  }

  @Override
  public String toString()  {
    StringBuilder sb = new StringBuilder();
    sb.append("class SkipTracingLookupResponseNamesAliases {\n");
    
    sb.append("  lastName: ").append(lastName).append("\n");
    sb.append("  middleName: ").append(middleName).append("\n");
    sb.append("  firstName: ").append(firstName).append("\n");
    sb.append("}\n");
    return sb.toString();
  }
}
